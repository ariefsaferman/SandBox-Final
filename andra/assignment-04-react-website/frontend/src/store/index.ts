import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import userReducer from "./slices/userSlice";
import thunk from "redux-thunk";

export const store = configureStore({
  reducer: {
    user: userReducer,
  },
  middleware: [logger, thunk],
});

export type RootState = ReturnType<typeof store.getState>;
