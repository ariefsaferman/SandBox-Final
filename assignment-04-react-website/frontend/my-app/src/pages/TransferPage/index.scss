.transfer__form {
  label {
    display: block;
    text-align-last: start;
  }

  .vertical {
    height: 45rem;
  }
}
