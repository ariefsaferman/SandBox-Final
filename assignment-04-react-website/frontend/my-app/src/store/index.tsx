import { configureStore } from "@reduxjs/toolkit";
import thunk from "redux-thunk";
import logger from "redux-logger";
import userReducer from "./slices/userSlice";
import transactionReducer from "./slices/transactionSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    transaction: transactionReducer,
  },
  middleware: [logger, thunk],
});

export type RootState = ReturnType<typeof store.getState>;
