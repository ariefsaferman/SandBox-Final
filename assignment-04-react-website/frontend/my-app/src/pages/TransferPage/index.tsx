import React, { useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import NavbarLogin from "../../components/Navbar";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../../store";
import { UserDispatch, fetchUser } from "../../store/slices/userSlice";

import "../style.scss";
import "./index.scss";
import { useCookies } from "react-cookie";
import { toast, ToastContainer } from "react-toastify";
import ModalDialog from "./components/ModalDialog";
import { ITransaction } from "../../interfaces/ITransactions";

interface TransferRequest {
  amount: number;
  to: number;
  description: string;
}

export default function TransferPage() {
  const API_URL = process.env.REACT_APP_API_URL;
  const { user } = useSelector((state: RootState) => state.user);
  const dispatch: UserDispatch = useDispatch();
  const [modal, setModal] = useState<boolean>(false);
  const [Tdata, setTdata] = useState<ITransaction>();

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<TransferRequest>();

  const [cookies] = useCookies(["token"]);

  const onSubmit = (data: TransferRequest) => {
    const transferRequest: TransferRequest = {
      amount: data.amount,
      to: data.to,
      description: data.description,
    };

    const requestOptions = {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${cookies.token}`,
      },
      body: JSON.stringify({
        amount: Number(transferRequest.amount),
        to: Number(transferRequest.to),
        description: transferRequest.description,
      }),
    };

    fetch(API_URL + "/transactions/transfer", requestOptions)
      .then((props) => {
        if (props.ok) {
          return props.json();
        }
        throw new Error("failed to transfer");
      })
      .then((res) => {
        if (res.data) {
          setModal(true);
          console.log(data);
          setTdata(res.data);
          dispatch(fetchUser(cookies.token));
        }
      })
      .catch((err) => {
        toast.error(err.message, {
          position: "top-right",
          autoClose: 5000,
          hideProgressBar: false,
          closeOnClick: true,
          pauseOnHover: true,
          draggable: true,
          progress: undefined,
          theme: "colored",
        });
      });
  };

  useEffect(() => {
    dispatch(fetchUser(cookies.token));
  }, [dispatch, cookies.token]);

  return (
    <div>
      <NavbarLogin />
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        theme="colored"
      />
      <ModalDialog showModal={modal} tData={Tdata} />

      <section className="transfer__form">
        <div className="container">
          <div className="d-flex vertical flex-column align-items-center justify-content-center">
            <h1 className="fw-bold">Transfer</h1>
            <form onSubmit={handleSubmit(onSubmit)}>
              <fieldset disabled>
                <label
                  htmlFor="exampleFormControlInput1"
                  className="form-label fw-bold mt-5"
                >
                  From
                </label>
                <div className="form-group mb-3">
                  <input
                    type="number"
                    placeholder={user?.id.toString()}
                    id="exampleFormControlInput1"
                    className="form-control control_login border border-dark"
                    aria-label="Username"
                    aria-describedby="basic-addon1"
                    style={{
                      width: "25rem",
                      height: "3rem",
                    }}
                  />
                </div>
              </fieldset>

              <label
                htmlFor="exampleFormControlInput2"
                className="form-label fw-bold "
              >
                To
              </label>
              <div className="form-group mb-3">
                <input
                  data-testid="toForm"
                  type="number"
                  id="exampleFormControlInput2"
                  className="form-control control_login border border-dark"
                  aria-label="Username"
                  aria-describedby="basic-addon1"
                  style={{
                    width: "25rem",
                    height: "3rem",
                    borderColor: errors.to ? "red" : "",
                  }}
                  {...register("to", { required: true })}
                />
                {errors.to && (
                  <span className="text-danger text-start">
                    Please insert destination wallet
                  </span>
                )}
              </div>

              <label
                htmlFor="exampleFormControlInput3"
                className="form-label fw-bold "
              >
                Amount
              </label>
              <div className="form-group mb-3">
                <input
                  type="number"
                  id="exampleFormControlInput3"
                  className="form-control control_login border border-dark"
                  aria-label="Username"
                  aria-describedby="basic-addon1"
                  placeholder="1.000.000"
                  style={{
                    width: "25rem",
                    height: "5rem",
                    fontSize: "2rem",
                    borderColor: errors.amount ? "red" : "",
                  }}
                  {...register("amount", {
                    required: true,
                    min: {
                      value: 1000,
                      message: "Minimal transfer Rp. 1.000",
                    },
                    max: {
                      value: 50000000,
                      message: "Maximal transfer Rp. 50.000.000",
                    },
                  })}
                />
                {errors.amount?.type === "required" ? (
                  <span data-testid="error" className="text-danger text-start">
                    Please insert amount
                  </span>
                ) : (
                  <span data-testid="error" className="text-danger text-start">
                    {errors.amount?.message}
                  </span>
                )}
              </div>

              <label
                htmlFor="exampleFormControlInput4"
                className="form-label fw-bold "
              >
                Description
              </label>
              <div className="form-group mb-3">
                <textarea
                  id="exampleFormControlInput4"
                  className="form-control control_login border border-dark"
                  aria-label="Username"
                  placeholder="Bayar Hutang"
                  aria-describedby="basic-addon1"
                  style={{
                    width: "25rem",
                    height: "5rem",
                    borderColor: errors.description ? "red" : "",
                  }}
                  {...register("description", { required: true })}
                />
                <span className="text-danger text-start">
                  {errors.description?.type === "required" ? (
                    <span className="text-danger text-start">
                      Please insert description
                    </span>
                  ) : (
                    <span className="text-danger text-start">
                      {errors.description?.message}
                    </span>
                  )}
                </span>
              </div>

              <div className="form-group mb-3 justify-content-center">
                <button
                  data-testid="submitBtn"
                  type="submit"
                  className="btn btn-info btn-lg w-100 text-white"
                >
                  Send
                </button>
              </div>
            </form>
          </div>
        </div>
      </section>
    </div>
  );
}
