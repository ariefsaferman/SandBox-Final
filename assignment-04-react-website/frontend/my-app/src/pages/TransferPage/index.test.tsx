import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import TransferPage from "./index";
import { CookiesProvider } from "react-cookie";
import { store } from "../../store";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import userEvent from "@testing-library/user-event";

type Props = {
  children: React.ReactNode;
};

test("Should return error when user input invalid to wallet", async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <TransferPage />
      </BrowserRouter>
    </Provider>
  );
  const screenTransfer = screen.getByTestId("toForm");
  await userEvent.click(screenTransfer);
  await userEvent.keyboard("0");
  fireEvent.click(screen.getByTestId("submitBtn"));
  const screenText = screen.getByTestId("error");
  expect(screenText).toBeInTheDocument();
});

const Providers = (props: Props) => {
  return (
    <CookiesProvider>
      <Provider store={store}>
        <BrowserRouter>{props.children}</BrowserRouter>
      </Provider>
    </CookiesProvider>
  );
};

test("render snapshot", () => {
  const view = render(<TransferPage />, { wrapper: Providers });
  expect(view).toMatchSnapshot();
});
