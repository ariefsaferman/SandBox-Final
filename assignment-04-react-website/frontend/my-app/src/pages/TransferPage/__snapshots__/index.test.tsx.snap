// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="home__page"
        >
          <div
            class="navbar navbar-expand-lg"
          >
            <div
              class="container"
            >
              <div
                class="navbar-brand fw-bold"
              >
                <h3>
                  DigiWallet
                </h3>
              </div>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                class="navbar-toggler"
                data-bs-target="#navbarSupportedContent"
                data-bs-toggle="collapse"
                type="button"
              >
                <span
                  class="navbar-toggler-icon"
                />
              </button>
              <div
                class="collapse navbar-collapse justify-content-end"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav "
                >
                  <li
                    class="nav-item ms-5"
                  >
                    <a
                      aria-current="page"
                      class="nav-link fs-5 active"
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="nav-item ms-5 "
                  >
                    <a
                      class="nav-link fs-5"
                      href="/transfer"
                    >
                      Transfer
                    </a>
                  </li>
                  <li
                    class="nav-item ms-5"
                  >
                    <a
                      class="nav-link fs-5"
                      href="/topup"
                    >
                      Topup
                    </a>
                  </li>
                  <li
                    class="nav-item ms-5"
                  >
                    <a
                      class="nav-link fs-5"
                      href="/games"
                    >
                      Games
                    </a>
                  </li>
                  <li
                    class="nav-item ms-5"
                  >
                    <a
                      class="nav-link fs-5"
                      href="/login"
                    >
                      Logout
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>
        <div
          class="Toastify"
        />
        <div>
          <button
            class="btn btn-primary"
            data-bs-target="#exampleModal"
            data-bs-toggle="modal"
            style="display: none;"
            type="button"
          >
            Primary
          </button>
          <div
            aria-hidden="true"
            aria-labelledby="exampleModalLabel"
            class="modal fade wrapper"
            data-bs-backdrop="static"
            data-bs-keyboard="false"
            id="exampleModal"
            tabindex="-1"
          >
            <div
              class="modal-dialog"
              style="margin-top: 11.5rem;"
            >
              <div
                class="modal-content"
              >
                <div
                  class="modal-header border-bottom-0"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="img__icon "
                      style="text-align: center;"
                    >
                      <svg>
                        success.svg
                      </svg>
                    </div>
                    <h1
                      class="modal-title text-success fw-bold"
                      style="text-align: center;"
                    >
                      Transfer Success
                    </h1>
                  </div>
                </div>
                <div
                  class="modal-body px-3"
                >
                  <table
                    class="table table-borderless"
                    style="width: 100%;"
                  >
                    <tbody>
                      <tr>
                        <td
                          style="text-align: left; font-size: 1.5rem; width: 50%;"
                        >
                          Amount
                        </td>
                        <td
                          style="text-align: right; font-size: 1.5rem; width: 50%;"
                        />
                      </tr>
                      <tr>
                        <td
                          style="text-align: left; font-size: 1.5rem; width: 50%;"
                        >
                          Transaction Id
                        </td>
                        <td
                          style="text-align: right; font-size: 1.5rem; width: 50%;"
                        />
                      </tr>
                      <tr>
                        <td
                          style="text-align: left; font-size: 1.5rem; width: 50%;"
                        >
                          From
                        </td>
                        <td
                          style="text-align: right; font-size: 1.5rem; width: 50%;"
                        />
                      </tr>
                      <tr>
                        <td
                          style="text-align: left; font-size: 1.5rem; width: 50%;"
                        >
                          To
                        </td>
                        <td
                          style="text-align: right; font-size: 1.5rem; width: 50%;"
                        />
                      </tr>
                      <tr>
                        <td
                          style="text-align: left; font-size: 1.5rem; width: 50%;"
                        >
                          Description
                        </td>
                        <td
                          style="text-align: right; font-size: 1.5rem; width: 50%;"
                        />
                      </tr>
                    </tbody>
                  </table>
                </div>
                <div
                  class="modal-footer justify-content-center gap-3 border-top-0 mb-4"
                >
                  <button
                    class="btn btn-outline-info fs-4"
                    style="width: 6rem;"
                    type="button"
                  >
                    Print
                  </button>
                  <button
                    class="btn btn-outline-info fs-4"
                    data-bs-dismiss="modal"
                    style="width: 6rem;"
                    type="button"
                  >
                    Close
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <section
          class="transfer__form"
        >
          <div
            class="container"
          >
            <div
              class="d-flex vertical flex-column align-items-center justify-content-center"
            >
              <h1
                class="fw-bold"
              >
                Transfer
              </h1>
              <form>
                <fieldset
                  disabled=""
                >
                  <label
                    class="form-label fw-bold mt-5"
                    for="exampleFormControlInput1"
                  >
                    From
                  </label>
                  <div
                    class="form-group mb-3"
                  >
                    <input
                      aria-describedby="basic-addon1"
                      aria-label="Username"
                      class="form-control control_login border border-dark"
                      id="exampleFormControlInput1"
                      style="width: 25rem; height: 3rem;"
                      type="number"
                    />
                  </div>
                </fieldset>
                <label
                  class="form-label fw-bold "
                  for="exampleFormControlInput2"
                >
                  To
                </label>
                <div
                  class="form-group mb-3"
                >
                  <input
                    aria-describedby="basic-addon1"
                    aria-label="Username"
                    class="form-control control_login border border-dark"
                    data-testid="toForm"
                    id="exampleFormControlInput2"
                    name="to"
                    style="width: 25rem; height: 3rem;"
                    type="number"
                  />
                </div>
                <label
                  class="form-label fw-bold "
                  for="exampleFormControlInput3"
                >
                  Amount
                </label>
                <div
                  class="form-group mb-3"
                >
                  <input
                    aria-describedby="basic-addon1"
                    aria-label="Username"
                    class="form-control control_login border border-dark"
                    id="exampleFormControlInput3"
                    name="amount"
                    placeholder="1.000.000"
                    style="width: 25rem; height: 5rem; font-size: 2rem;"
                    type="number"
                  />
                  <span
                    class="text-danger text-start"
                    data-testid="error"
                  />
                </div>
                <label
                  class="form-label fw-bold "
                  for="exampleFormControlInput4"
                >
                  Description
                </label>
                <div
                  class="form-group mb-3"
                >
                  <textarea
                    aria-describedby="basic-addon1"
                    aria-label="Username"
                    class="form-control control_login border border-dark"
                    id="exampleFormControlInput4"
                    name="description"
                    placeholder="Bayar Hutang"
                    style="width: 25rem; height: 5rem;"
                  />
                  <span
                    class="text-danger text-start"
                  >
                    <span
                      class="text-danger text-start"
                    />
                  </span>
                </div>
                <div
                  class="form-group mb-3 justify-content-center"
                >
                  <button
                    class="btn btn-info btn-lg w-100 text-white"
                    data-testid="submitBtn"
                    type="submit"
                  >
                    Send
                  </button>
                </div>
              </form>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="home__page"
      >
        <div
          class="navbar navbar-expand-lg"
        >
          <div
            class="container"
          >
            <div
              class="navbar-brand fw-bold"
            >
              <h3>
                DigiWallet
              </h3>
            </div>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse justify-content-end"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav "
              >
                <li
                  class="nav-item ms-5"
                >
                  <a
                    aria-current="page"
                    class="nav-link fs-5 active"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item ms-5 "
                >
                  <a
                    class="nav-link fs-5"
                    href="/transfer"
                  >
                    Transfer
                  </a>
                </li>
                <li
                  class="nav-item ms-5"
                >
                  <a
                    class="nav-link fs-5"
                    href="/topup"
                  >
                    Topup
                  </a>
                </li>
                <li
                  class="nav-item ms-5"
                >
                  <a
                    class="nav-link fs-5"
                    href="/games"
                  >
                    Games
                  </a>
                </li>
                <li
                  class="nav-item ms-5"
                >
                  <a
                    class="nav-link fs-5"
                    href="/login"
                  >
                    Logout
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
      <div
        class="Toastify"
      />
      <div>
        <button
          class="btn btn-primary"
          data-bs-target="#exampleModal"
          data-bs-toggle="modal"
          style="display: none;"
          type="button"
        >
          Primary
        </button>
        <div
          aria-hidden="true"
          aria-labelledby="exampleModalLabel"
          class="modal fade wrapper"
          data-bs-backdrop="static"
          data-bs-keyboard="false"
          id="exampleModal"
          tabindex="-1"
        >
          <div
            class="modal-dialog"
            style="margin-top: 11.5rem;"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header border-bottom-0"
              >
                <div
                  class="col"
                >
                  <div
                    class="img__icon "
                    style="text-align: center;"
                  >
                    <svg>
                      success.svg
                    </svg>
                  </div>
                  <h1
                    class="modal-title text-success fw-bold"
                    style="text-align: center;"
                  >
                    Transfer Success
                  </h1>
                </div>
              </div>
              <div
                class="modal-body px-3"
              >
                <table
                  class="table table-borderless"
                  style="width: 100%;"
                >
                  <tbody>
                    <tr>
                      <td
                        style="text-align: left; font-size: 1.5rem; width: 50%;"
                      >
                        Amount
                      </td>
                      <td
                        style="text-align: right; font-size: 1.5rem; width: 50%;"
                      />
                    </tr>
                    <tr>
                      <td
                        style="text-align: left; font-size: 1.5rem; width: 50%;"
                      >
                        Transaction Id
                      </td>
                      <td
                        style="text-align: right; font-size: 1.5rem; width: 50%;"
                      />
                    </tr>
                    <tr>
                      <td
                        style="text-align: left; font-size: 1.5rem; width: 50%;"
                      >
                        From
                      </td>
                      <td
                        style="text-align: right; font-size: 1.5rem; width: 50%;"
                      />
                    </tr>
                    <tr>
                      <td
                        style="text-align: left; font-size: 1.5rem; width: 50%;"
                      >
                        To
                      </td>
                      <td
                        style="text-align: right; font-size: 1.5rem; width: 50%;"
                      />
                    </tr>
                    <tr>
                      <td
                        style="text-align: left; font-size: 1.5rem; width: 50%;"
                      >
                        Description
                      </td>
                      <td
                        style="text-align: right; font-size: 1.5rem; width: 50%;"
                      />
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="modal-footer justify-content-center gap-3 border-top-0 mb-4"
              >
                <button
                  class="btn btn-outline-info fs-4"
                  style="width: 6rem;"
                  type="button"
                >
                  Print
                </button>
                <button
                  class="btn btn-outline-info fs-4"
                  data-bs-dismiss="modal"
                  style="width: 6rem;"
                  type="button"
                >
                  Close
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <section
        class="transfer__form"
      >
        <div
          class="container"
        >
          <div
            class="d-flex vertical flex-column align-items-center justify-content-center"
          >
            <h1
              class="fw-bold"
            >
              Transfer
            </h1>
            <form>
              <fieldset
                disabled=""
              >
                <label
                  class="form-label fw-bold mt-5"
                  for="exampleFormControlInput1"
                >
                  From
                </label>
                <div
                  class="form-group mb-3"
                >
                  <input
                    aria-describedby="basic-addon1"
                    aria-label="Username"
                    class="form-control control_login border border-dark"
                    id="exampleFormControlInput1"
                    style="width: 25rem; height: 3rem;"
                    type="number"
                  />
                </div>
              </fieldset>
              <label
                class="form-label fw-bold "
                for="exampleFormControlInput2"
              >
                To
              </label>
              <div
                class="form-group mb-3"
              >
                <input
                  aria-describedby="basic-addon1"
                  aria-label="Username"
                  class="form-control control_login border border-dark"
                  data-testid="toForm"
                  id="exampleFormControlInput2"
                  name="to"
                  style="width: 25rem; height: 3rem;"
                  type="number"
                />
              </div>
              <label
                class="form-label fw-bold "
                for="exampleFormControlInput3"
              >
                Amount
              </label>
              <div
                class="form-group mb-3"
              >
                <input
                  aria-describedby="basic-addon1"
                  aria-label="Username"
                  class="form-control control_login border border-dark"
                  id="exampleFormControlInput3"
                  name="amount"
                  placeholder="1.000.000"
                  style="width: 25rem; height: 5rem; font-size: 2rem;"
                  type="number"
                />
                <span
                  class="text-danger text-start"
                  data-testid="error"
                />
              </div>
              <label
                class="form-label fw-bold "
                for="exampleFormControlInput4"
              >
                Description
              </label>
              <div
                class="form-group mb-3"
              >
                <textarea
                  aria-describedby="basic-addon1"
                  aria-label="Username"
                  class="form-control control_login border border-dark"
                  id="exampleFormControlInput4"
                  name="description"
                  placeholder="Bayar Hutang"
                  style="width: 25rem; height: 5rem;"
                />
                <span
                  class="text-danger text-start"
                >
                  <span
                    class="text-danger text-start"
                  />
                </span>
              </div>
              <div
                class="form-group mb-3 justify-content-center"
              >
                <button
                  class="btn btn-info btn-lg w-100 text-white"
                  data-testid="submitBtn"
                  type="submit"
                >
                  Send
                </button>
              </div>
            </form>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
